substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/pthread/PthreadVMLockSupport.java

```diff
@@ -95,11 +95,14 @@ public void duringSetup(DuringSetupAccess access) {
 
     @Override
     public void beforeCompilation(BeforeCompilationAccess access) {
+        // This should actually be: `max(alignof(pthread_mutex_t), alignof(pthread_cond_t))`.
+        // Until `alignof()` can be queried from the C compiler, we hard-code this to two words.
+        final int alignment = 2 * ConfigurationValues.getTarget().wordSize;
```

Comment:
The comment is good.  Can we hard-code to `8` instead?  Then it wouldn't introduce any layout change on 64bit platforms.

Reply:
I still don't want to hard-code this with numbers. Please take a look at my new changes.